src/o365/spo/commands/web/web-set.ts

```diff
@@ -102,6 +103,26 @@ class SpoWebSetCommand extends SpoCommand {
 
     request
       .patch(requestOptions)
+      .then((): Promise<void> => {
+        if (typeof args.options.welcomePage === 'undefined') {
+          return Promise.resolve();
+        }
+        if (this.verbose) {
+          cmd.log(`Updating Welcome page for the site ${args.options.webUrl}`);
+        }
+        const requestOptions: any = {
+          url: `${args.options.webUrl}/_api/web/rootfolder`,
+          headers: {
+            'Content-Type': 'application/json;odata=nometadata',
+            accept: 'application/json;odata=nometadata',
+            'IF-MATCH': '*',
+            'X-HTTP-Method': 'PATCH'
+          },
+          __metadata: {'type': 'SP.Folder'},
```

Comment:
If `__metadata` were to be specified, they should go in the body rather than request options. However, because the content-type is set to `nometadata`, specifying metadata is not necessary and can be removed.